json-endpoints/class.wpcom-json-api-export-start-endpoint.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+// POST /sites/%s/exports/start
+class WPCOM_JSON_API_Export_Start_Endpoint extends WPCOM_JSON_API_Endpoint {
+	protected $needed_capabilities = 'export';
+
+	function callback( $path = '', $blog_id = 0 ) {
+		$blog_id = $this->api->switch_to_blog_and_validate_user( $this->api->get_blog_id( $blog_id ) );
+		if ( is_wp_error( $blog_id ) ) {
+			return $blog_id;
+		}
+
+		if ( !current_user_can( $this->needed_capabilities ) ) {
+			return new WP_Error( 'unauthorized', 'User cannot export', 403 );
+		}
+		$args = $this->input();
+
+		// Set up args array
+		$args = $this->setup_args( $args );
+
+		/**
+		 * Filters the export args.
+		 *
+		 * @since 3.5.0
+		 *
+		 * @param array $args The arguments to send to the exporter.
+		 */
+		$args = apply_filters( 'export_args', $args );
+
+		// Using output buffering to catch the exporter response and write it to a file.
+		include( ABSPATH . 'wp-admin/includes/export.php' );
+		ob_start();
+		export_wp( $args);
+		$string_data = ob_get_clean();
+
+		// Export file is saved to the uploads folder.
+		$file_name = '/export_' . wp_create_nonce() . '.xml';
+		$file_path = wp_upload_dir()['path'] . $file_name;
+		file_put_contents( $file_path, $string_data );
+
+		$file_url = wp_upload_dir()['url'] . $file_name;
+		return array(
+			'status'        => 'success',
+			'download_url'  => $file_url
+		);
+	}
+
+	/**
+	 * setup_args is the glue between the wpcom API request and the .org exporter.
+	 * It translates the parameters for partial exports from one format to the other.
+	 *
+	 * @param $args - Parameters received from the API request
+	 *
+	 * @return array - Parameters for the .org exporter
+	 */
+	private function setup_args( $args ) {
+		$prepared_args = array();
+
+		/**
+		 * Export options.
+		 *
+		 */
+
+		// Export everything
+		if ( empty( $args['post_type'] ) || 'all' === $args['post_type'] ) {
```

Comment:
Would assigning the defaults be better accomplished with [`wp_parse_args`](https://codex.wordpress.org/Function_Reference/wp_parse_args)?